pysatMadrigal/instruments/methods/madrigal.py

```diff
@@ -436,14 +436,13 @@ def get_remote_filenames(inst_code=None, kindat=None, user=None,
                 date_array))
         start = date_array.min()
         stop = date_array.max()
-        if start == stop:
-            stop += dt.timedelta(days=1)
-
+    if start == stop:
+        stop += dt.timedelta(days=1)
```

Comment:
Regardless of whether date-array is specified, we need to check if a single day is requested and adjust accordingly.